cmd/az-analyze/cmd/common_test.go

```diff
@@ -0,0 +1,170 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"time"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/fake"
+	v1beta1 "sigs.k8s.io/azuredisk-csi-driver/pkg/apis/azuredisk/v1beta1"
+	diskfakes "sigs.k8s.io/azuredisk-csi-driver/pkg/apis/client/clientset/versioned/fake"
+	consts "sigs.k8s.io/azuredisk-csi-driver/pkg/azureconstants"
+)
+
+func NewTestK8sClientset() *fake.Clientset {
+
+	return fake.NewSimpleClientset(&v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test-pod-0",
+			Namespace: "default",
+		},
+		Spec: v1.PodSpec{
+			Volumes: []v1.Volume{
+				{
+					Name: "volume-name-0",
+					VolumeSource: v1.VolumeSource{
+						PersistentVolumeClaim: &v1.PersistentVolumeClaimVolumeSource{
+							ClaimName: "test-pvcClaimName-0",
+						},
+					},
+				},
+				{
+					Name: "volume-name-1",
+					VolumeSource: v1.VolumeSource{
+						PersistentVolumeClaim: &v1.PersistentVolumeClaimVolumeSource{
+							ClaimName: "test-pvcClaimName-1",
+						},
+					},
+				},
+			},
+		},
+	}, &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test-pod-1",
+			Namespace: "default",
+		},
+		Spec: v1.PodSpec{
+			Volumes: []v1.Volume{
+				{
+					Name: "volume-name",
+					VolumeSource: v1.VolumeSource{
+						PersistentVolumeClaim: &v1.PersistentVolumeClaimVolumeSource{
+							ClaimName: "test-pvcClaimName-0",
+						},
+					},
+				},
+			},
+		},
+	}, &v1.Node{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   "test-node-0",
+			Labels: map[string]string{consts.WellKnownTopologyKey: "eastus-0"},
+		},
+	}, &v1.Node{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   "test-node-1",
+			Labels: map[string]string{consts.WellKnownTopologyKey: "eastus-1"},
+		},
+	})
+}
+
+func NewTestAzDiskClientset() *diskfakes.Clientset {
+	return diskfakes.NewSimpleClientset(&v1beta1.AzVolume{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test-azVolume-0",
+			Namespace: consts.DefaultAzureDiskCrdNamespace,
+		},
+		Spec: v1beta1.AzVolumeSpec{
+			VolumeName: "test-azVolume-0",
+			Parameters: map[string]string{consts.PvcNameKey: "test-pvcClaimName-0"},
+		},
+		Status: v1beta1.AzVolumeStatus{
+			State: v1beta1.VolumeCreated,
+		},
+	}, &v1beta1.AzVolume{
```

Comment:
It would be cleaner to create slices for AzVolume and AzVolumeAttachment separately and then add those to this function. Otherwise, if we want to add more objects later, this definition will keep getting bigger.